### PR TITLE
update fcm conversion branch and address PR feedback

### PR DESCRIPTION
**Pull request by ehaynes99**
This is a fork of:
https://github.com/banshiAnton/node-pushnotifications/tree/banshiAnton-fcm-method

That branch has an open PR here: https://github.com/appfeel/node-pushnotifications/pull/194 The original author has not responded in months, and this conversion is necessary before the removal of the legacy firebase API in June.

I don't take credit for any of the prior work. I simply merged the current master and addressed the outstanding PR feedback here: https://github.com/appfeel/node-pushnotifications/pull/194#issuecomment-2088770323

### CODE REVIEW

File: test/send/FCM-service-account-key.json
Comment (miqmago) [inline test/send/FCM-service-account-key.json:1]: I'm not sure if this is private info?

**Review by miqmago — COMMENTED** [review]
Just a quick question about a file that seems private info

File: test/send/sendFCM.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    'sendEachForMulticast',
+    function sendFCM(firebaseMessage) {
+      const { custom, ...messageData } = message;
+
+      expect(firebaseMessage.tokens).to.deep.equal(regIds);
+
+      expect(firebaseMessage.android.priority).to.equal('high');
+      expect(firebaseMessage.android.notification).to.deep.include(messageData);
+
+      expect(firebaseMessage.apns.payload.aps.sound).to.equal(
```
Comment (jamesbluecrow) [inline test/send/sendFCM.js:46]: This should be `firebaseMessage.android.payload.aps.sound`

File: test/send/sendFCM.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        messageData.sound
+      );
+
+      expect(firebaseMessage.apns.payload.aps.alert).to.deep.include({
```
Comment (jamesbluecrow) [inline test/send/sendFCM.js:50]: This should be `firebaseMessage.android.payload.aps. alert`

File: test/send/sendFCM.js (hunk 1 above)
Comment (ehaynes99) [reply test/send/sendFCM.js:46]: Not according to the types: https://github.com/firebase/firebase-admin-node/blob/a37eb6cb5ad92d911d0b868f5beb8d11761bcc0a/src/messaging/messaging-api.ts#L20

the `android?: AndroidConfig` does not have a `payload` property, but `apns?: ApnsConfig;` does, which has a `aps: Aps;` that contains a `sound?: string | CriticalSound;` property.

File: test/send/sendFCM.js (hunk 2 above)
Comment (ehaynes99) [reply test/send/sendFCM.js:50]: Not according to the types: https://github.com/firebase/firebase-admin-node/blob/a37eb6cb5ad92d911d0b868f5beb8d11761bcc0a/src/messaging/messaging-api.ts#L20

the `android?: AndroidConfig` does not have a `payload` property, but `apns?: ApnsConfig;` does, which has a `aps: Aps;` that contains a `alert?: string | ApsAlert;` property.

**Comment by NicolasBonduel** [comment]
Could you add @alex-friedl as a reviewer?

**Comment by ehaynes99** [comment]
Actually, sorry, I thought the original PR was in a better state, but we've found several issues. Also, this is *really* suspicious: https://github.com/appfeel/node-pushnotifications/blob/2f6c2685f343816cb93edb573e828609e355d3ef/src/sendFCM.js#L83

I'm not comfortable submitting this anymore. I'm going to convert our app to use `firebase-admin` directly. I'll leave this branch around for a bit if anyone wants to fork it.

**Comment by Hossman333** [comment]
@jamesbluecrow or @NicolasBonduel Any clarification on the concern @ehaynes99 brought up? 🤔 👀

**Comment by jamesbluecrow** [comment]
> @jamesbluecrow or @NicolasBonduel Any clarification on the concern @ehaynes99 brought up? 🤔 👀

Nope, I agree with @ehaynes99 (we work together). 

We tested the PR and it didn't work as expected and we found a couple of bugs. 

We decided to just move out of this library and use `firebase-admin-sdk` instead.

**Pull request closed without merging**